src/main/java/org/jabref/gui/filelist/AttachFileAction.java

```diff
@@ -2,43 +2,61 @@
+import org.jabref.gui.DialogService;
+import org.jabref.gui.actions.SimpleCommand;
 import org.jabref.gui.undo.UndoableFieldChange;
+import org.jabref.gui.util.FileDialogConfiguration;
 import org.jabref.logic.l10n.Localization;
 import org.jabref.model.FieldChange;
 import org.jabref.model.entry.BibEntry;
 import org.jabref.model.entry.FieldName;
 import org.jabref.model.entry.LinkedFile;
+import org.jabref.preferences.JabRefPreferences;
 
-public class AttachFileAction implements BaseAction {
+public class AttachFileAction extends SimpleCommand {
 
     private final BasePanel panel;
+    private final DialogService dialogService;
 
-    public AttachFileAction(BasePanel panel) {
+    public AttachFileAction(BasePanel panel, DialogService dialogService) {
         this.panel = panel;
+        this.dialogService = dialogService;
     }
 
     @Override
-    public void action() {
+    public void execute() {
         if (panel.getSelectedEntries().size() != 1) {
             panel.output(Localization.lang("This operation requires exactly one item to be selected."));
             return;
         }
         BibEntry entry = panel.getSelectedEntries().get(0);
-        LinkedFile flEntry = new LinkedFile("", "", "");
-        FileListEntryEditor editor = new FileListEntryEditor(flEntry, false, true,
-                panel.getBibDatabaseContext());
-        editor.setVisible(true, true);
-        if (editor.okPressed()) {
-            Optional<FieldChange> fieldChange = entry.addFile(flEntry);
-
-            if (fieldChange.isPresent()) {
-                UndoableFieldChange ce = new UndoableFieldChange(entry, FieldName.FILE,
-                        entry.getField(FieldName.FILE).orElse(null), fieldChange.get().getNewValue());
-                panel.getUndoManager().addEdit(ce);
-                panel.markBaseChanged();
-            }
-        }
+
+        FileDialogConfiguration fileDialogConfiguration = new FileDialogConfiguration.Builder()
+                                                                                               .withInitialDirectory(Globals.prefs.get(JabRefPreferences.WORKING_DIRECTORY))
+                                                                                               .build();
+
+        dialogService.showFileOpenDialog(fileDialogConfiguration).ifPresent(newFile -> {
+            LinkedFile newLinkedFile = new LinkedFile("", newFile.toString(), "");
+
+            LinkedFilesWrapper wrapper = new LinkedFilesWrapper();
+            wrapper.setLinkedFile(newLinkedFile);
+            LinkedFileEditDialogView dialog = new LinkedFileEditDialogView(wrapper);
+
+            Optional<LinkedFile> editedLinkeDfile = dialog.showAndWait();
+            editedLinkeDfile.ifPresent(file -> {
+
+                Optional<FieldChange> fieldChange = entry.addFile(file);
+
+                if (fieldChange.isPresent()) {
+                    UndoableFieldChange ce = new UndoableFieldChange(entry, FieldName.FILE,
```

Comment:
There should be constructor of `UndoableFieldChange` that directly accepts a `FieldChange`. If not, please add one and use it here.